File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for brac in s:
        if not stack :
            if brac == "(" : stack.append("(")
            else : return False
        
        else:
            if brac == "(" : stack.append("(")
            else : stack.pop()
            
    if stack : return False            
    else : return True 
    
    
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let stack  = []
    
    s.split("").forEach((bracket,idx) => {
        if(bracket === "(") stack.push(bracket)
        else{
            if (stack.length >= 1) {
                stack.pop()
            }
            else {
                stack.push(")")
                return false
            }
        }
    })
    return stack.length === 0 ? true : false;
}